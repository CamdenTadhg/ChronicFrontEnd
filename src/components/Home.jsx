import React from 'react';
import {useSelector} from 'react-redux';
import {Link} from 'react-router-dom';
import {selectProfile} from '../redux/selectors/selectors';
import {Col, Row, Container} from 'reactstrap';
import LatestSidebar from './LatestSidebar';
import logo from '../assets/Chronic_dark.png';

function Home() {
    const profile = useSelector(selectProfile);

    return (
        <Container fluid className='p-0'>
            <Row className='navbar-space m-0 p-0'>
            </Row>
            <Row className='main-body m-0 p-0'>
                <Col className='home' xs='9'>
                    {profile.userId ? 
                        <div className='in-home-page'>
                            <Link to='/tracking' className='link-box-a'><div className='link-box'>Tracking</div></Link>
                            <div className='central-directions'>
                                <div className='directions'>
                                    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={5} stroke="currentColor" className="arrow">
                                    <path strokeLinecap="round" strokeLinejoin="round" d="M10.5 19.5 3 12m0 0 7.5-7.5" />
                                    <path strokeLinecap="butt" strokeLinejoin="round" d="M3 12h18"/></svg>
                                    <div className='directions-text'>To track your symptoms and medications, click here</div>
                                </div>
                                <div className='directions'>
                                    <div className='directions-text'>To see your tracking data in a graph, click here</div>
                                    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={5} stroke="currentColor" className="arrow">
                                    <path strokeLinecap="round" strokeLinejoin="round" d="M13.5 4.5 21 12m0 0-7.5 7.5" />
                                    <path strokeLinecap='butt' strokeLinejoin='round' d='M21 12H3'/></svg>
                                </div>
                            </div>
                            <Link to='/data' className='link-box-a'><div className='link-box'>Data</div></Link>
                        </div>
                        : 
                        <div className='out-home-page'>
                            <div className='hero-text'>Your Health. Your Data. Your Power.</div>
                            <img className='hero-image' src={logo} alt='Logo'></img>
                            <div className='hero-text'>CHRONIC</div>

                        </div>
                    }
                </Col>
                <Col className='sidebar' xs='3'>
                    <LatestSidebar/>
                </Col>
            </Row>
        </Container>

    )
}

export default Home;