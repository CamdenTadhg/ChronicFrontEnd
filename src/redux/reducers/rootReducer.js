import {combineReducers} from 'redux';
import profileReducer from './profileReducer';
import trackingReducer from './trackingReducer';
import dataReducer from './dataReducer';
import latestReducer from './latestReducer';

const rootReducer = combineReducers({
    profile: profileReducer,
    tracking: trackingReducer,
    data: dataReducer,
    latest: latestReducer
});

export default rootReducer;