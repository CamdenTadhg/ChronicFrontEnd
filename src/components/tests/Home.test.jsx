import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react';
import {it, expect, vi} from 'vitest';
import * as router from 'react-router';
import '@testing-library/jest-dom';
import {MemoryRouter} from 'react-router-dom';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import Home from '../Home';
import rootReducer from '../../redux/reducers/rootReducer';
import ChronicAPI from '../../api/chronicAPI';

const navigate = vi.fn();
vi.spyOn(router, 'useNavigate').mockImplementation(() => navigate);

it('renders without errors', () => {
    const store = configureStore({reducer:rootReducer});
    render(
        <Provider store={store}>
            <MemoryRouter>
                <Home/>
            </MemoryRouter>
        </Provider>
    )
});

it('matches the snapshot', () => {
    const store = configureStore({reducer: rootReducer});
    const home = render(
        <Provider store={store}>
            <MemoryRouter>
                <Home/>
            </MemoryRouter>
        </Provider>
    );
    expect(home).toMatchSnapshot();
});

it('displays the appropriate content to an anonymous user', () => {
    const store = configureStore({reducer: rootReducer});
    const {getByText} = render(
        <Provider store={store}>
            <MemoryRouter>
                <Home/>
            </MemoryRouter>
        </Provider>
    );
    expect(getByText('Your Health. Your Data. Your Power.')).toBeInTheDocument();
});

it('displays the appropriate context to a logged in user', async () => {
    const store = configureStore({reducer: rootReducer, preloadedState: {
        profile: {
            userId: 7,
            diagnoses: [{
                diagnosis: 'chronic fatigue syndrome', 
                keywords: ['ME']
            }]
        }
    }});
    ChronicAPI.getArticleIds = vi.fn();
    ChronicAPI.getArticleIds.mockResolvedValue([1234,2233]);
    ChronicAPI.getArticles = vi.fn();
    ChronicAPI.getArticles.mockResolvedValue([
        {PMID: 2345, title: 'a test ME article', abstract: 'this is a test article about ME'},
        {PMID: 5732, title: 'a second test ME article', abstract: 'this is also a test article about ME'}
    ]);
    const {getByText} = render(
        <Provider store={store}>
            <MemoryRouter>
                <Home/>
            </MemoryRouter>
        </Provider>
    );
    await waitFor(() => expect(getByText('To see your tracking data in a graph, click here')).toBeInTheDocument());
});

it('allows navigation to tracking and data pages', async () => {
    const store = configureStore({reducer: rootReducer, preloadedState: {
        profile: {
            userId: 7,
            diagnoses: [{
                diagnosis: 'chronic fatigue syndrome', 
                keywords: ['ME']
            }]
        }
    }});
    ChronicAPI.getArticleIds = vi.fn();
    ChronicAPI.getArticleIds.mockResolvedValue([1234,2233]);
    ChronicAPI.getArticles = vi.fn();
    ChronicAPI.getArticles.mockResolvedValue([
        {PMID: 2345, title: 'a test ME article', abstract: 'this is a test article about ME'},
        {PMID: 5732, title: 'a second test ME article', abstract: 'this is also a test article about ME'}
    ]);
    const {getByText} = render(
        <Provider store={store}>
            <MemoryRouter>
                <Home/>
            </MemoryRouter>
        </Provider>
    );
    fireEvent.click(getByText('Tracking'));
    await waitFor(() => expect(navigate.mock.calls[0][0]).toBe('/tracking'));
    window.history.back();
    fireEvent.click(getByText('Data'));
    await waitFor(() => expect(navigate.mock.calls[1][0]).toBe('/data'));
});