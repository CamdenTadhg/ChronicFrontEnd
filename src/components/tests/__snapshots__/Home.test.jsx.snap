// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-0 container-fluid"
      >
        <div
          class="navbar-space m-0 p-0 row"
        />
        <div
          class="main-body m-0 p-0 row"
        >
          <div
            class="home col-9"
          >
            <div
              class="out-home-page"
            >
              <div
                class="hero-text"
              >
                Your Health. Your Data. Your Power.
              </div>
              <img
                alt="Logo"
                class="hero-image"
                src="/src/assets/Chronic_dark.png"
              />
              <div
                class="hero-text"
              >
                CHRONIC
              </div>
            </div>
          </div>
          <div
            class="sidebar col-3"
          >
            <a
              href="/latest"
            >
              <h2
                class="latest-head"
              >
                The Latest
              </h2>
            </a>
            <div
              class="articles-sidebar"
            >
              <p>
                Loading...
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-0 container-fluid"
    >
      <div
        class="navbar-space m-0 p-0 row"
      />
      <div
        class="main-body m-0 p-0 row"
      >
        <div
          class="home col-9"
        >
          <div
            class="out-home-page"
          >
            <div
              class="hero-text"
            >
              Your Health. Your Data. Your Power.
            </div>
            <img
              alt="Logo"
              class="hero-image"
              src="/src/assets/Chronic_dark.png"
            />
            <div
              class="hero-text"
            >
              CHRONIC
            </div>
          </div>
        </div>
        <div
          class="sidebar col-3"
        >
          <a
            href="/latest"
          >
            <h2
              class="latest-head"
            >
              The Latest
            </h2>
          </a>
          <div
            class="articles-sidebar"
          >
            <p>
              Loading...
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
